firstvoices/backend/tests/test_async/test_alphabet_tasks.py

```diff
@@ -0,0 +1,100 @@
+        assert result == {
+            "unknown_character_count": {"⚑a": 1, "⚑b": 1, "⚑c": 1},
+            "updated_entries": [],
+        }
+
+    @pytest.mark.django_db
+    def test_recalulate_preview_updated_only(self, site, alphabet):
+        factories.DictionaryEntryFactory.create(site=site, title="abc")
+        factories.CharacterFactory.create(site=site, title="a")
+        factories.CharacterFactory.create(site=site, title="b")
+        factories.CharacterFactory.create(site=site, title="c")
+
+        result = recalculate_custom_order_preview("test")
+        assert result == {
+            "unknown_character_count": {},
+            "updated_entries": [
+                {
+                    "title": "abc",
+                    "cleaned_title": "abc",
+                    "new_custom_order": "!#$",
+                    "previous_custom_order": "⚑a⚑b⚑c",
+                }
+            ],
+        }
+
+    @pytest.mark.django_db
+    def test_recalulate_preview_confusables_only(self, site, alphabet):
+        factories.CharacterFactory.create(site=site, title="A")
+        factories.DictionaryEntryFactory.create(site=site, title="ᐱᐱᐱ")
+        alphabet.input_to_canonical_map = [{"in": "ᐱ", "out": "A"}]
+        alphabet.save()
+
+        result = recalculate_custom_order_preview("test")
+        assert result == {
+            "unknown_character_count": {},
+            "updated_entries": [
+                {
+                    "title": "ᐱᐱᐱ",
+                    "cleaned_title": "AAA",
+                    "new_custom_order": "!!!",
+                    "previous_custom_order": "⚑ᐱ⚑ᐱ⚑ᐱ",
+                }
+            ],
+        }
+
+    @pytest.mark.django_db
+    def test_recalulate_preview_unknown_updated_confusables(self, site, alphabet):
+        factories.CharacterFactory.create(site=site, title="A")
+        factories.DictionaryEntryFactory.create(site=site, title="ᐱᐱᐱ")
+        factories.DictionaryEntryFactory.create(site=site, title="abcd")
+        factories.CharacterFactory.create(site=site, title="a")
+        factories.CharacterFactory.create(site=site, title="b")
+        factories.CharacterFactory.create(site=site, title="c")
+        alphabet.input_to_canonical_map = [{"in": "ᐱ", "out": "A"}]
+        alphabet.save()
+
+        result = recalculate_custom_order_preview("test")
+        assert result == {
+            "unknown_character_count": {"⚑d": 1},
+            "updated_entries": [
+                {
+                    "title": "abcd",
+                    "cleaned_title": "abcd",
+                    "new_custom_order": "#$%⚑d",
+                    "previous_custom_order": "⚑a⚑b⚑c⚑d",
+                },
+                {
+                    "title": "ᐱᐱᐱ",
+                    "cleaned_title": "AAA",
+                    "new_custom_order": "!!!",
+                    "previous_custom_order": "⚑ᐱ⚑ᐱ⚑ᐱ",
+                },
+            ],
```

Comment:
I think we need a couple more test cases:

- a dictionary entry that had no unknown characters, and is unaffected by the recalculate
- a dictionary entry that does have unknown characters, and is unaffected by the recalculate
- a dictionary entry that had no unknown characters, and is changed by the recalculate